web/src/components/core/MenuButton.tsx

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) [2025] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useId, useMemo, useRef, useState } from "react";
+import {
+  MenuToggle,
+  DrilldownMenu,
+  MenuContent,
+  Divider,
+  MenuContainer,
+  Menu,
+  MenuList,
+  MenuItem,
+  MenuPopperProps,
+  MenuItemProps,
+  MenuToggleProps,
+} from "@patternfly/react-core";
+import { _ } from "~/i18n";
+
+/** PF complains with ids generated by useId because they contain colons. Such ids cannot be used
+ * for menus.
+ */
+function useMenuId() {
+  const id = useId();
+  const menuId = useMemo(() => id.replaceAll(":", ""), [id]);
+  return menuId;
+}
+
+interface MenuButtonItemProps extends Omit<MenuItemProps, "direction" | "drilldownMenu"> {
+  items?: React.ReactNode[];
+  upProps?: { label?: string };
+}
+
+export function MenuButtonItem({
+  items = [],
+  upProps = { label: _("Back") },
+  children,
+  ...props
+}: React.PropsWithChildren<MenuButtonItemProps>): React.ReactNode {
+  const itemId = useId();
+  const menuId = useMenuId();
+  const onClick = (callback) => {
+    return (e) => {
+      e.stopPropagation();
+      callback && callback();
+    };
+  };
+
+  if (!items.length) {
+    props = { itemId, ...props };
+    return <MenuItem {...props}>{children}</MenuItem>;
+  }
+
+  props = { itemId, ...props, onClick: onClick(props.onClick) };
+
+  return (
+    <MenuItem
+      {...props}
+      direction="down"
```

Comment:
Why not destructuring the onClick instead?

```suggestion
export function MenuButtonItem({
  items = [],
  upProps = { label: _("Back") },
  onClick,
  children,
  ...props
}: React.PropsWithChildren<MenuButtonItemProps>): React.ReactNode {
  const itemId = useId();
  const menuId = useMenuId();
  const onItemClick = (event) => {
    event.stopPropagation();
    onClick && onClick(event);
  };

  if (!items.length) {
    return <MenuItem itemId={itemId} {...props}>{children}</MenuItem>;
  }
  return (
    <MenuItem
      itemId={itemId}
      {...props}
      onClick={onItemClick}
      direction="down"
```

I also took the opportunity for not overwriting props and avoid destructuring it twice.

In any case, I have no strong opinion here, although I feel getting ride of the callback thing make it easier to follow. Let me know if I'm missing something.

Reply:
I like your proposal. Please, apply it ;-).